### ISSUE TITLE
for-loop not properly inspected for some conditions

### ISSUE THREAD

**Issue by fredricj** [kalessil/phpinspectionsea#664]
The following code is tagged by suspicious loop inspection with warning "Condition and repeated operation are not complimentary..."

```
$s = 50;
$z = [];
for ($i = 0; $s > $i; $i++) {
	$z[] = $i;
}
```
These 2 functionally equivalent examples does not trigger the same warning:
```
$z2 = [];
for ($i = 0; 50 > $i; $i++) {
	$z2[] = $i;
}

$s = 50;
$z3 = [];
for ($i = 0; $i < $s; $i++) {
	$z3[] = $i;
}
```

**Comment by kalessil** [comment]
Looks like a bug, thank you for reporting @fredricj
